source/libraries/CompilerConfiguration.ts

```diff
@@ -0,0 +1,24 @@
+import * as path from 'path';
+
+export class CompilerConfiguration {
+    public readonly contractSourceRoot: string;
+    public readonly outputRoot: string;
+    public readonly contractInterfacesOutputPath: string;
+    public readonly abiOutputPath: string
+    public readonly contractOutputPath: string
+
+    public constructor(contractSourceRoot: string, outputRoot: string) {
+        this.contractSourceRoot = contractSourceRoot;
+        this.outputRoot = outputRoot;
+        this.contractInterfacesOutputPath = path.join(contractSourceRoot, '../libraries', 'ContractInterfaces.ts');
+        this.abiOutputPath = path.join(outputRoot, 'abi.json');
+        this.contractOutputPath = path.join(outputRoot, 'contracts.json');
+    }
+
+    public static create(): CompilerConfiguration {
```

Comment:
Recommend removing this method and moving its contents to the constructor.  The reason there was a factory previously was because some of the configuration options were created asynchronously and you can't have an async constructor.